backend/src/main/kotlin/com/github/quillraven/toilapp_dataloader/DataLoaderApplication.kt

```diff
@@ -0,0 +1,144 @@
+import org.springframework.stereotype.Component
+import reactor.core.publisher.Mono
+import java.io.InputStream
+import java.util.*
+import java.util.concurrent.Callable
+import kotlin.random.Random
+import kotlin.system.exitProcess
+
+val titles = listOf(
+    "Beautiful Toilet", "Shit Heaven", "Plumber's Paradies", "Flipi's World", "Diarrhea nightmares", "Decent public toilet", "When you gotta you gott go"
+)
+val descriptions = listOf(
+        "Very nice loo.",
+        "No toilet paper :( :(",
+        "Only on layer toilet paper :(",
+        "Very nice pooping experience.",
+        "One of the most beautiful toilets in the neighborhood.",
+        "Best crap ever!"
+)
+val commentTexts = listOf( "It was a pleasure", "Pleasant shit. I'll come back for sure.")
+val userNames = listOf(
+        "user1",
+        "namibian",
+        "specialist",
+        "homes",
+        "imaginary",
+        "weathertop",
+        "banana",
+        "plink",
+        "manrope,",
+        "rapid",
+        "moonraker",
+        "wholesale",
+        "rundown",
+        "movement",
+        "partridge",
+        "obsessed",
+        "pannikin"
+)
+
+@Configuration
+@EnableAutoConfiguration
+@EnableReactiveMongoRepositories(basePackages = ["com.github.quillraven.toilapp.repository"])
+@ComponentScan(
+        basePackages = ["com.github.quillraven.toilapp.model",  "com.github.quillraven.toilapp.repository",  "com.github.quillraven.toilapp.service",
+            "com.github.quillraven.toilapp_dataloader"]
+)
+class DataLoaderApplication
+
+@Component
+class DataLoaderRunner(
+        @Autowired private val imgService: ImageService,
+        @Autowired private val toiletService: ToiletService,
+        @Autowired private val userService: UserService,
+        @Autowired private val context: ConfigurableApplicationContext
+): CommandLineRunner {
+
+    val numToilets = 20
+
+    override fun run(vararg args: String?) {
+        createUsers().flatMap { createToilets(it) }.subscribe{
+            println("$numToilets toilets created..")
+            exitProcess(SpringApplication.exit(context))
+        }
+    }
+
+    private fun createUsers(): Mono<List<Any>> {
+        val userMonoList = mutableListOf<Mono<User>>()
+        //create users
+        for(usrName in userNames) {
+            val userName = userNames[Random.nextInt(0, userNames.size)]
+            val email = "$userName@mail.com"
+            val user = User(UUID.randomUUID().toString(), userName, email)
+            userMonoList.add(userService.createUser(user))
+        }
+        return Mono.zip(userMonoList){ uArr -> uArr.toList() }
+    }
+
+    private fun createToilets(uList: List<Any>): Mono<List<Any>> {
+        val toiletMonoList = mutableListOf<Mono<Toilet>>()
+        for(num in 0 until numToilets) {
+            toiletMonoList.add(createToilet(uList))
+        }
+        return Mono.zip(toiletMonoList){ tArr -> tArr.toList()}
```

Comment:
I am not 100% sure about the difference of `Mono.zip` and this code

```kotlin
 return Flux
            .fromIterable(userMonoList)
            .flatMap { it }
            .collectList()
```

With the code provided you do not lose the type information and the result is therefore a `Mono<MutableList<User>>` which can be returned bis this function to avoid some casts later on. 

It seems to work for me but again I am not 100% sure what is the difference.